app/src/main/java/com/courseproject/tindar/usecases/chat/ChatInteractor.java

```diff
@@ -0,0 +1,99 @@
+package com.courseproject.tindar.usecases.chat;
+
+import com.courseproject.tindar.ds.DatabaseHelper;
+import com.courseproject.tindar.entities.MessageModel;
+
+import java.util.ArrayList;
+
+public class ChatInteractor implements ChatPresenter, ChatActivityController{
+    private final ChatDsGateway chatDsHelper;
+    private final String userId;
+    private String conversationPartnerUserId;
+    private String[] userIds;
+    private String conversationId;
+    private ArrayList<MessageModel> messageList;
+    private final ArrayList<ChatPresenter> observers;
+
+    public ChatInteractor(ChatDsGateway chatDsHelper, String userId,
```

Comment:
Interactor is just for interacting with other dependency injection. userId or other stuff can be handled in the method